Makefile

```diff
@@ -0,0 +1,167 @@
+VERSIONDATE = 2021/01/25
+# user callable NEW option, to specify the new version. If unspecified, the
+# new version gets determined by git.
+ifdef NEW
+    VERSIONNEXT = $(NEW)
+else
+    VERSIONNEXT = $(shell git describe --tags --dirty)
+endif
+VERSIONDATENEXT = $(shell date +"%Y\/%m\/%d")
+TARBALL=moderncv-$(VERSIONNEXT).tar
+
+EXAMPLESDIR = examples
+# MANUALTEX is used by the userguide method that operates in the 
+# $(MANUALDIR) folder. MANUAL is used by methods operating in 
+# the $(MODERNCVDIR) folder. This is to ensuer that the userguide 
+# is also buildable within the manual folder.
+MANUALTEX = moderncv_userguide.tex
+MANUAL = $(MANUALDIR)/$(MANUALTEX)
+TEMPLATE = $(MODERNCVDIR)/template.tex
+TEMPLATEBIB = publications.bib
+TEMPLATEBASE = $(basename $(TEMPLATE))
+TEMPLATEPDF = $(addsuffix .pdf,$(TEMPLATEBASE))
+MANUAL_BASE = $(basename $(MANUAL))
+MANUALPDF = $(addsuffix .pdf,$(MANUAL_BASE))
+
+# redefine the template rule depending on whether the user has specified STYLE
+# or not.
+ifdef STYLE
+	# in this case user has specified STYLE
+else
+	STYLE = casual
+endif
+template: $(TEMPLATE) $(TEMPLATEBIB)
+	if [[ $(strip $(STYLE)) == "casual" ]]; then
+		# build template in default style
+		latexmk -pdflua -bibtex -quiet "$<"
+	else
+		# build template in style $(STYLE). This assumes that casual is the default.
+		sedstring="s/moderncvstyle{casual}/moderncvstyle{$(STYLE)}/g"
+		sed -i $$sedstring $(TEMPLATE)
+		# build template in specified style
+		latexmk -pdflua -bibtex -quiet "$<"
+		# reset template to default value
+		sedstring="s/moderncvstyle{$(STYLE)}/moderncvstyle{casual}/g"
+		sed -i $$sedstring $(TEMPLATE)
+	fi
+
+
+templates: $(TEMPLATE) $(TEMPLATEBIB)
+	# build the template for each style and store pdfs in the examples folder
+	# this is done to include these expamples in release tar balls.
+	mkdir -p $(EXAMPLESDIR)
+	previousstyle="casual"
+	for style in casual classic banking oldstyle fancy; do
+		sedstring="s/moderncvstyle{$$previousstyle}/moderncvstyle{$$style}/g"
+		sed -i $$sedstring $(TEMPLATE)
+		latexmk -pdflua -bibtex -quiet $(TEMPLATE)
+		cp $(TEMPLATEPDF) $(EXAMPLESDIR)/$(TEMPLATEBASE)-$$style.pdf
+		mv $(TEMPLATEPDF) $(MANUALDIR)/$(TEMPLATEBASE)-$$style.pdf
+		previousstyle=$$style
+		unset sedstring
+	done
+	sedstring="s/moderncvstyle{$$previousstyle}/moderncvstyle{casual}/g"
+	sed -i $$sedstring $(TEMPLATE)
+
+
+userguide: templates $(MANUAL)
+	# build the user guide. Since the guide includes the template examples, we
+	# build those first by calling the templates rule.
+	cd $(MANUALDIR)
+	./format_files_for_userguide.py 
+	lualatex $(MANUALTEX)
+	lualatex $(MANUALTEX)
+	cd ..
+
+
+version:
+	# Upate version information and date of all moderncv files. A call make version
+	# will define VERSIONNEXT=$(shell git describe --tags). Alternatively, call
+	# "make version NEW=v5.13.298" to set v5.13.298 as the new version number.
+	# The date gets calculated by the date function.
```

Comment:
Ok so this just doesn't work in this order. We first need to update the files, and then tag it via git. The other way round is completely useless (sorry). The version should be set by the user, and then git tagged.

Reply:
No. This rule automates updating all files, that's the point. The makefile provides this method and the VERSION variable to make it easy to update all files in one go. From the explanation of the README:

`version:` Update the version information (version number and date) of all `moderncv` files (*.sty, moderncv.cls, *.tex). This rule can be called in two different ways. Note, however, that it is intended to be called by the rule `release` and usually does not need to be called explicitly.
  * `make version:` Called in this way the version number is obtained through `git describe --tags`. If this information is newer all `moderncv` files get updated. 
  * `make version NEW=<version number>:` Optionally, the desired version number `<version number>` can be specified. 

I first intended this rule to be called in the second way by the maintainer to first update all files as you said. 
Then I thought it might be usefull to automate this with git, so that as a mainteiner you only need to run one command at the end and all information is up to date automatically.

Reply:
I was intending to reduce your work load for releasing tarball to CTAN ;-)

Reply:
> `make version`: Called in this way the version number is obtained through git describe --tags. If this information is newer all moderncv files get updated.

Yes exactly that one doesn't make sense, since this will always be outdated. First the change of version number, then the git tag.